supplementary/alica_ros1/alica_tests/src/test/Expr/include/alica_tests/GoalPlan.h

```diff
@@ -3,46 +3,62 @@
 #include <alica_tests/DomainCondition.h>
 #include <alica_tests/DomainPlan.h>
 #include <boost/dll/alias.hpp>
+#include <engine/BasicConstraint.h>
 #include <engine/BasicUtilityFunction.h>
 #include <engine/DefaultUtilityFunction.h>
 #include <engine/UtilityFunction.h>
 
 namespace alica
 {
+class ProblemDescriptor;
+
 class GoalPlan : public DomainPlan
 {
 public:
     GoalPlan(PlanContext& context);
 };
 
-class PreCondition1402489131988 : public DomainCondition
-{
-    bool evaluate(std::shared_ptr<RunningPlan> rp, const Blackboard* gb);
-};
-class RunTimeCondition1403773741874 : public DomainCondition
-{
-    bool evaluate(std::shared_ptr<RunningPlan> rp, const Blackboard* gb);
-};
-class PreCondition1402489174338 : public DomainCondition
+class GoalPlanPreCondition : public DomainCondition
 {
 public:
+    GoalPlanPreCondition(ConditionContext& context)
+            : DomainCondition(){};
     bool evaluate(std::shared_ptr<RunningPlan> rp, const Blackboard* gb);
+    static std::unique_ptr<GoalPlanPreCondition> create(ConditionContext& context) { return std::make_unique<GoalPlanPreCondition>(context); };
 };
-class PreCondition1402489206278 : public DomainCondition
+class GoalPlanRuntimeCondition : public DomainCondition
 {
 public:
+    GoalPlanRuntimeCondition(ConditionContext& context)
+            : DomainCondition(){};
     bool evaluate(std::shared_ptr<RunningPlan> rp, const Blackboard* gb);
+    static std::unique_ptr<GoalPlanRuntimeCondition> create(ConditionContext& context) { return std::make_unique<GoalPlanRuntimeCondition>(context); };
 };
-class PreCondition1402489218027 : public DomainCondition
+class GoalPlanPostCondition : public DomainCondition
```

Comment:
we can get rid of DomainCondition entirely & direcly inherit here from BasicCondition